refresh/repo.go

```diff
@@ -41,11 +43,11 @@ func DefaultRefreshTokenGenerator() ([]byte, error) {
 type RefreshTokenRepo interface {
 	// Create generates and returns a new refresh token for the given client-user pair.
 	// On success the token will be return.
-	Create(userID, clientID string) (string, error)
+	Create(userID, clientID string, scope []string) (string, error)
 
-	// Verify verifies that a token belongs to the client, and returns the corresponding user ID.
+	// Verify verifies that a token belongs to the client and is valid for the given scopes. It returns the user ID to which the token belongs.
 	// Note that this assumes the client validation is currently done in the application layer,
-	Verify(clientID, token string) (string, error)
+	Verify(clientID, token string, scopes []string) (string, error)
```

Comment:
Why does this takes a "scopes" argument? I wouldn't expect those to change between successive refresh requests.